Simbody/tests/TestParallelForces.cpp

```diff
@@ -0,0 +1,168 @@
+public:
+    mutable bool hasRealized[Stage::Report+1];
+    NonParallelForceImpl() {
+        for (int i = 0; i < Stage::NValid; i++)
+            hasRealized[i] = false;
+    }
+    bool shouldBeParallelIfPossible() const override{
+      return false;
+    }
+    void calcForce(const State& state, Vector_<SpatialVec>& bodyForces, Vector_<Vec3>& particleForces, Vector& mobilityForces) const {
+         std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    Real calcPotentialEnergy(const State& state) const {
+        return 0.0;
+    }
+    void realizeTopology(State& state) const {
+        hasRealized[Stage::Topology] = true;
+    }
+    void realizeModel(State& state) const {
+        hasRealized[Stage::Model] = true;
+    }
+    void realizeInstance(const State& state) const {
+        hasRealized[Stage::Instance] = true;
+    }
+    void realizeTime(const State& state) const {
+        hasRealized[Stage::Time] = true;
+    }
+    void realizePosition(const State& state) const {
+        hasRealized[Stage::Position] = true;
+    }
+    void realizeVelocity(const State& state) const {
+        hasRealized[Stage::Velocity] = true;
+    }
+    void realizeDynamics(const State& state) const {
+        hasRealized[Stage::Dynamics] = true;
+    }
+    void realizeAcceleration(const State& state) const {
+        hasRealized[Stage::Acceleration] = true;
+    }
+    void realizeReport(const State& state) const {
+        hasRealized[Stage::Report] = true;
+    }
+};
+
+void testParallelForce()
+{
+    MultibodySystem system;
+    SimbodyMatterSubsystem matter(system);
+    GeneralForceSubsystem forces(system);
+    
+    for(int x = 0; x < 50; x++)
+      Force::Custom custom(forces, new ParallelForceImpl());
+    
+    system.realizeTopology();
+    State state = system.getDefaultState();
+    system.realize(state, Stage::Dynamics);
+}
+
+void testNonParallelForce()
+{
+    MultibodySystem system;
+    SimbodyMatterSubsystem matter(system);
+    GeneralForceSubsystem forces(system);
+    
+    for(int x = 0; x < 50; x++)
+      Force::Custom custom(forces, new NonParallelForceImpl());
+    
+    system.realizeTopology();
+    State state = system.getDefaultState();
+    system.realize(state, Stage::Dynamics);
+}
+
+int main()
+{
+    auto start = std::chrono::high_resolution_clock::now();
+    testParallelForce();
+    auto end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double, std::milli> elapsedParallel = end-start;
+    
+    start = std::chrono::high_resolution_clock::now();
+    testNonParallelForce();
+    end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double, std::milli> elapsedNonParallel = end-start;
+    
+    cout << "Parallel Forces Time: " << elapsedParallel.count() << endl;
+    cout << "NonParallel Forces Time: " << elapsedNonParallel.count() << endl;
+    
+    assert(elapsedNonParallel.count() < elapsedParallel.count());
```

Comment:
This seems backwards! Also, `assert` is a macro that is only present in Debug builds. You can use SimTK_ASSERT_ALWAYS. Or, better to use the SimTK testing framework. [Here](https://github.com/simbody/simbody/blob/master/SimTKcommon/tests/TestCloneOnWritePtr.cpp#L432) is an example.